Automattic/jetpack
# E2E tests: better scoping

## Proposed changes:

Update e2e tests scoping to use jetpack cli dependencies command and run tests whenever any dependency of the specified target plugin changes.

- moved the target definition from `package.json` to `e2e-matrix.js` file for more control. You can now define a group of tests to run multiple times for different conditions.
- for monorepo changes there is one small group to run, just to validate the infrastructure didn't break
- for changes to files that have no tests assigned, test report and slack notifications jobs no longer run

### Other information:


## Jetpack product discussion
pd5faL-pt-p2#comment-308

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* [see test run](https://github.com/Automattic/jetpack/actions/runs/4162842667/jobs/7202505998) for changes to monorepo
* [see test run](https://github.com/Automattic/jetpack/actions/runs/4162815533) for changes to files without tests assigned
* [see test run](https://github.com/Automattic/jetpack/actions/runs/4162613454/jobs/7202023879) for changes to only boost plugin
